use-cases/inferencing/serving-with-vllm/README.md

```diff
@@ -0,0 +1,513 @@
+
+By the end of this guide, you should be able to perform the following steps:
+
+- Create a Persistent Disk for the LLM model weights
+- Deploy a vLLM container to your cluster to host your model
+- Use vLLM to serve the fine-tuned Gemma model
+- View Production metrics for your model serving
+- Use custom metrics and Horizontal Pod Autoscaler (HPA) to scale your model
+
+## Prerequisites
+
+- This guide was developed to be run on the [playground AI/ML platform](/platforms/gke-aiml/playground/README.md). If you are using a different environment the scripts and manifest will need to be modified for that environment.
+- A bucket containing the fine-tuned model from the [Fine-tuning example](/use-cases/model-fine-tuning-pipeline/fine-tuning/pytorch/README.md)
+
+## Preparation
+
+- Clone the repository and change directory to the guide directory
+
+  ```sh
+  git clone https://github.com/GoogleCloudPlatform/accelerated-platforms && \
+  cd accelerated-platforms/use-cases/inferencing/serving-with-vllm
+  ```
+
+- Ensure that your `MLP_ENVIRONMENT_FILE` is configured
+
+  ```sh
+  cat ${MLP_ENVIRONMENT_FILE} && \
+  source ${MLP_ENVIRONMENT_FILE}
+  ```
+
+  > You should see the various variables populated with the information specific to your environment.
+
+- Set environment variables
+
+  ```sh
+  PROJECT_ID=your-project-id>
+  PROJECT_NUMBER=$(gcloud projects describe ${PROJECT_ID} --format="value(projectNumber)")
+  V_MODEL_BUCKET=<model-artifacts-bucket>
+  CLUSTER_NAME=<your-gke-cluster>
+  NAMESPACE=ml-team
+  KSA=<k8s-service-account>
+  MODEL_ID=<your-model-id>
+  REGION=<your-region>
+  IMAGE_NAME=<your-image-name>
+  DISK_NAME=<your-disk-name>
+  ZONE=<your-disk-zone>
+  ACCELERATOR_TYPE=<accelerator_type> # nvidia-l4 | nvidia-tesla-a100
+  ```
+
+- Get Credentials for the GKE cluster
+
+  ```sh
+  gcloud container fleet memberships get-credentials ${MLP_CLUSTER_NAME} --project ${MLP_PROJECT_ID}
+  ```
+
+- Grant permission to kubernetes service account in cluster to access the storage bucket to view model weights
+
+  ```sh
+  gcloud storage buckets add-iam-policy-binding "gs://$V_MODEL_BUCKET" \
+    --member "principal://iam.googleapis.com/projects/"${PROJECT_NUMBER}"/locations/global/workloadIdentityPools/${PROJECT_ID}.svc.id.goog/subject/ns/$NAMESPACE/sa/$KSA" \
+    --role "roles/storage.objectViewer"
+  ```
+
+- Update the bucket access level to uniform.
+
+  ```sh
+  gcloud storage buckets update "gs://$V_MODEL_BUCKET"  --uniform-bucket-level-access
+  ```
+
+## Create PV, PVC and Persistent disk
+
+Loading model weights from a Persistent Volume is a method to load models faster. In GKE, Persistent Volumes backed by GCP Persistent Disks can be mounted read-only simultaneously by multiple nodes(ReadOnlyMany), this allows multiple pods access the model weights from a single volume.
+
+  Create a PVC for the model weights
```

Comment:
should we be creating these in the `default` namespace?

Reply:
We need a separate namespace for hosting the model for security and functionality reasons. In my opinion, enterprises would prefer to host the model at least in a separate namespace and hence restrict all the resource access to the serving namespace only.

Reply:
the PVC would end up in the model serving namespace, but the job etc would likely be in a operator/gitops namespace to prepare the data, but not default